ui/src/pages/feature/useGraphHook.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect } from "react";
+import { Elements } from 'react-flow-renderer';
+import { generateEdge, generateNode } from "../../components/graph/utils";
+import { FeatureLineage } from "../../models/model";
+
+const useGraphHook = (lineageData: FeatureLineage, featureType: string|null, setElements: Function) => {
+  // Generate graph data on client side, invoked after graphData or featureType is changed
+  useEffect(() => {
```

Comment:
useEffect is a lifecycle method which should be explicitly used in component. Then the scope of this helper function can be reduced only do data transformation without any UI logic.

There is already a graph utils class under graph folder, please move this helper function there.